File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.java
class Solution {
    public int solution(int[] arr1, int[] arr2) {
        int answer = 0; //정답 판별
        int result1 = 0;
        int result2 = 0;
        
        if(arr1.length != arr2.length) { //배열의 길이가 다른 경우
            if(arr1.length > arr2.length) {
                answer = 1;
            } else {
                answer = -1;
            }
        } else { //배열의 길이가 같은 경우
            for(int i = 0; i < arr1.length; i++) {
                result1 += arr1[i];
                result2 += arr2[i];
            }
            if(result1 > result2) {
                answer = 1;
            } else if(result1 < result2) {
                answer = -1;
            } else {
                answer = 0;
            }
        }        
        return answer;
    }
}

File: 프로그래머스/0/181856. 배열 비교하기/배열 비교하기.js
function solution(arr1, arr2) {
    var answer = 0;
    var sum1 = 0; // arr1요소합
    var sum2 = 0; // arr2요소합
    
    for(let i of arr1) {
        sum1 += i;
    }
    
    for(let i of arr2) {
        sum2 += i;
    }
    
    if(arr1.length != arr2.length) {
        if(arr1.length > arr2.length) {
            answer = 1;
        } else {
            answer = -1;
        }
    } else {
        if(sum1 > sum2) {
            answer = 1;
        } else if(sum1 < sum2) {
            answer = -1;
        } else {
            answer = 0;
        }
    }
    
    return answer;
}